### DISCUSSION TITLE
Be able to tag any building as demolished or destroyed

### DISCUSSION THREAD

**Discussion by ancarda** [streetcomplete/StreetComplete#2975] [Ideas]
**Use case**
The other day, a friend of mine saw a local business burn down. He subsequently updated Google Maps to indicate the business is temporally closed, but wasn't able to figure out how to make the same edit on OpenStreetMap. I had a look on the wiki and wasn't entirely sure how to tag it either.

As far as I can tell, you change this:

```
amenity=restaurant
building=yes
```

To this:

```
destroyed:amenity=restaurant
destroyed:building=yes
```

**Proposed Solution**
I think StreetComplete might be a good app to be able to make these edits. I imagine you could tap on any building and mark it as demolished or destroyed. For destroyed, perhaps it could ask you how (fire and earthquake?), and add the relevant tags. This wiki page: https://wiki.openstreetmap.org/wiki/Key:destroyed: seems to suggest `damage:type` might exist, but is not documented?

I appreciate this is a rough idea, and maybe needs some standardization on the wiki first. I'm not sure where else to ask.

**Comment by matkoniecz** [comment]
>  I imagine you could tap on any building and mark it as demolished or destroyed.

For such editing I would recommend Vespucci. StreetComplete is intentionally offering only predefined quests, not a full scale editing.

**Comment by smichel17** [comment]
I'm going to convert this to a discussion since I am quite sure that the feature as-is would be rejected as out-of-scope, but is still interesting to discuss, so I don't want to close it outright.

**Comment by smichel17** [comment]
I've seen a number of requests like this, for some "full-fledged" editor feature. Just now, I realized that I think they point to a **missing niche in the OSM ecosystem**. Or rather, two niches which are both filled only by StreetComplete: directed and non-technical. Not sure those are the best names, but here's what I mean:

- **Directed:** StreetComplete detects missing tags and surfaces those to the user using quests, in a prioritized order. Compare with other editors, where you kind of need to know what you're opening the app to do.
- **Non-technical:** StreetComplete maps natural language answers to tags, so you don't have to learn anything about OSM tagging schemas.

It's easy to conflate these qualities because they both contribute to StreetComplete's ease of use. However, they *are* distinct. You could imagine an editor with quests that prompt the user to enter the tag value in free-form text. Or an editor which which supports applying some limited set of actions to any element without knowing the tags, but is totally user-directed; this one could include your suggestion here and many similar ones which are out of scope for StreetComplete (in my opinion, see below).

The problem is that **expanding functionality makes directed apps less usable.** Since, by definition, they can't rely on users searching out some specific functionality, each feature needs to get the user's attention if it's going to get used (ref: https://github.com/streetcomplete/StreetComplete/issues/2970). But, add too many and once again the user has to explore them and decide which one to use, restoring the learning curve and defeating the purpose of being a directed app. Good design can mitigate this, but this is hard work and there is still a hard limit.

Worth noting that in the past @westnordost and I have not seen totally eye-to-eye on this; see https://github.com/streetcomplete/StreetComplete/issues/2461, starting around https://github.com/streetcomplete/StreetComplete/issues/2461#issuecomment-756093959. Perhaps this time I've explained myself well enough to convince you :stuck_out_tongue_winking_eye:

There's nothing wrong with undirected apps. They can support many more features without becoming unusable. You just can't be directed and undirected at once. However, StreetComplete receives feature requests for both types of features, because it is the only popular non-technical OSM editor. So, I think the missing link is a **non-technical, undirected OSM editor.** It would have features like this suggestion, adding new buildings or other geometry, and editing tags on any element, but would be limited to a set of tags and values.

There's probably more I could say, but I'll leave it here.

**Reply by mnalis** [reply]
@smichel17 I very much agree with your definition of two distinctive features of SC. However, as for the conclusions I'm not so sure:

- apps can (kind of) be both directed and undirected. For example SC is mostly directed, but it has one feature that is undirected (long press to `create new Note` - which I happen to use **a lot**). While it could be applied to direct OSM editing (long press to manually edit nodes/ways), I find such idea wasteful. It would either need to be:
  - limited (like, SC implementing `long press here to force re-asking all quests on this node/way` and thus miss many features -  like the one that prompted this discussion), 
  - or it would be reimplementing Vespucci inside of SC, which is IMHO a bad idea.
 Afterall, I can also in SC long press and choose `Open location in another app` (like Vespucci) and have less complexity, and more choice/more features.

- I'm not sure that (another) **non-technical, undirected OSM editor** is needed. Maybe it is, but Vespucci (again, kind of) offers this already. 
 If you edit the node/way in Vespucci, you can press icon `+Abc` which adds **only** tags useful for that feature, and shown in **natural language** (not tag-techical-keynames) like `Entrance width in cm`, `Capacity (overall)`, `Spaces for disabled`, `Street name`, `Roof type`, `Access` etc. and also provides fixed values for those that has list of values  (like `Public` / `Private` / `No access` / `Permit required` for access, or `Gabled` / `Hipped` etc. **even with pictures** for roof type). 

  Sure, Vespucci **also** allows for free-form tag editing for advanced users who know technical tags by heart. But maybe there is some mode when advanced usage is disabled by default (I don't know all Vespucci features) and only such preconfigured answers are allowed like in `+Abc`. If not, it would be simple adding of `Mode: beginner / xxx / advanced` to it instead of creating new 99% similar app.

**Reply by smichel17** [reply]
> apps can (kind of) be both directed and undirected.

:100: — Like *most* ways of categorizing things, there are nice, clear theoretical boundaries that become blurry when they meet the real world; they're still useful concepts, as long as you don't apply them too dogmatically. Even *highly* complex, undirected tools tend to have a directed tutorial (e.g. `vimtutor`).

With the right design, you can also cram a lot of undirected power into an otherwise-directed tool with a minimal effect on its learning curve (at least, the beginning of it :wink:). Particularly with scriptable tools. My favorite example of this is **[SimpleTask](https://github.com/mpcjanssen/simpletask-android)**. While there is still a ***lot*** of room for improvement to reach it, the idea is that most of the UI is directed, with a limited amount of customizability… but, there's a scripting system that lets you customize a lot of the behavior. This means the UI doesn't need to offer every option under the sun — it stays _simple_.

For example, you can give tasks a threshold date, and there's a gui toggle to hide tasks until their threshold date is passed. I think that's the only visibility option related to dates. But if you want, you could write a script that shows tasks past the threshold *or* within 1 week of the due date. Or to hide tasks until N days after their threshold date, where N is the number of characters in the task. Why you'd want to do that, I haven't a clue, but it's possible with very little complication of the gui — mostly just a textbox.

> Vespucci (again, kind of) offers this already.

Well, uh, I can't help but **[agree](https://github.com/streetcomplete/StreetComplete/issues/2461#issuecomment-756093959)** with you.

Really, I can't comment too much here. When I used Vespucci **several years ago,** it was quite at the extreme of undirected and I did not have the patience/motivation to figure out how to do the few minor edits I wanted it for. The learning curve may have changed since then :)

**Reply by westnordost** [reply]
> non-technical, undirected OSM editor. 

I plan to do this, with those "Layers". The feature will be "hidden" enough (in the menu) that normal SC users will not be bothered by it.

**Comment by mnalis** [comment]
As for the @ancarda I would suggest just long-pressing on the location, choosing `Create new note`, and typing down "this restaurant has burned to the ground", maybe adding a picture. Then other mappers (or even yourself, in leisure, with Wiki, mailing lists and other communication methods at your disposal if you wish to learn and do it yourself) can tag it appropriately.

There are literally millions of things that can be wrong on the map, and IMHO adding a millions of quests is not the solution if SC is to remain usable. OSM **Notes**, especially when accompanied by pictures as SC allows, are universal free-form way to mark any and every inconsistency on the map. Especially as burned down houses are (hopefully) something that happens very infrequently.
